More changes (mainly for Streaming and Relion)

delarosatrevin opened the pull request:
### Summary 

#### Relion protocols
* Fixed bug in Relion autopick protocol when downscaling in streaming
* Allow to provide references for 2D and 3D classification
* Propagated rlnRandomSubset, needed by Localrec
* Bug fixed in Relion particles-extraction when using not integer scale
* Added protocol in Relion to center averages (center of mass in relion_image_handler)
* Added protocol to export-particles in Relion format
* Allow to pass the calibrated pixel size in Relion postprocess protocol

#### Streaming
* Defined 'sleep time' for some quick protocols in streaming. This reduce the IO operations for updates checking.
* Re-factoring Gautomatch to use bad coords in streaming
* Gctf can use now multi GPUs
* Gautomatch can use multi GPUs and batch mode
* Relion auto-pick can use batch steps
* Relion extract-particles re-factored, now in with batch mode and unified with non-streaming.

#### Others
* Allow to merge sets with different attributes
* Use double-click to select in Dialogs (more intuitive)
* Allow to quickly rename a protocol label (rigth-click -> Rename option)
* CTFModel now allows to have phaseShift information as part of the model
* Picking wizard now allows to pick all micrographs at once (more efficient in some programs)

delarosatrevin committed:
Bug fixed in Relion particles-extraction when using not integer scale factor

josuegbl committed:
remove motioncor2 v1.0.0 and v1.0.4 from install

shalinis602 committed:
Set micName when importing relion particles

delarosatrevin committed:
Propagated rlnRandomSubset, needed by Localrec

delarosatrevin committed:
Updated labels and help for Relion mask 3d, added test

delarosatrevin committed:
Some code formating in Relion 3d classify

delarosatrevin committed:
Defining a sleeping time in streaming for ctf and picking

pconesa committed:
Allow join sets of different attributtes getting only the common ones. Warn about it.

pconesa committed:
Merge branch 'facilities' into fixes

pconesa committed:
Handle phaseshift:
For relion from and to Scipion
Ignore phaseshift for xmipp, since it is not used.
Test particle conversion for relion.

delarosatrevin committed:
Make double-click select items on ListDialog

pconesa committed:
Only PEP8 changes
Some commented code removed.

pconesa committed:
Move getAttributesToStoreRecursive to Object

delarosatrevin committed:
Store last event coordinates

delarosatrevin committed:
Propagate the **kwargs and react to position in Dialog

delarosatrevin committed:
Allow to rename protocol label from protocol tree/list

delarosatrevin committed:
Set to True references inverted contrast, since most common in Cryo

pconesa committed:
CTFModel may or may not have phaseShift attribute to avoid persisting extra values.
Converts of ctffind, gCtf, relion, Xmipp adapted to this.
Test: relion convert and ctf_streaming avid use of CTFModel._phaseShift.

delarosatrevin committed:
Re-factoring Gautomatch to use bad coords in streaming

pconesa committed:
Remove unnecessary funtion in Object and use existing one

delarosatrevin committed:
Several changes to manage the Automatic state (used in GenericPicker)...with help from Airen

delarosatrevin committed:
Merge branch 'sdevel' of github.com:delarosatrevin/scipion into sdevel

delarosatrevin committed:
Some formatting and added validation of second level attrs when joining

pconesa committed:
Merge pull request #1 from delarosatrevin/join_sets_bug_jm

Some formatting and added validation of second level attrs when joining

delarosatrevin committed:
Merge pull request #1496 from pconesa/join_sets_bug

Allow join sets of different attributtes getting only the common ones…
Extended the test to check the fix and all related subset tests passed.

josuegbl committed:
changed the way to treat phaseshifts to CTFModels

delarosatrevin committed:
Merge branch 'release-1.1.facilities-devel' into sdevel

josuegbl committed:
Merge branch 'release-1.1.facilities-devel' into release-1.1.facilities-fixes

delarosatrevin committed:
Fixed bug introduced when changing the convert.py

delarosatrevin committed:
Merge branch 'release-1.1.facilities-fixes' into sdevel

delarosatrevin committed:
Fixed bug in Relion autopick protocol when downscaling in streaming

delarosatrevin committed:
Adding parameters for classification references

delarosatrevin committed:
Merge pull request #1497 from I2PC/release-1.1.facilities-fixes

All test related to CTF passed! Good work @josuegbl :+1:

delarosatrevin committed:
Merge branch 'sdevel' of github.com:delarosatrevin/scipion into sdevel

delarosatrevin committed:
Merge branch 'release-1.1.facilities-devel' into sdevel

josuegbl committed:
test modified to support phaseShitf

josuegbl committed:
Merge branch 'release-1.1.facilities-devel' into release-1.1.facilities-fixes

josuegbl committed:
Merge pull request #1510 from I2PC/release-1.1.facilities-fixes

Release 1.1.facilities fixes

pconesa committed:
Allow instantiation of CTFModel with optional _phaseShift kwargs

delarosatrevin committed:
Merge branch 'release-1.1.facilities-fixes' into sdevel

delarosatrevin committed:
Picked gctf for particles from @azazellochg implementation

delarosatrevin committed:
Fixed condition for referenceAverages param

delarosatrevin committed:
Fixed bug in Executor with when threads are less than gpus

josuegbl committed:
Merge pull request #1518 from I2PC/release-1.1.facilities-fixes

Allow instantiation of CTFModel with optional _phaseShift kwargs

azazellochg committed:
bugfix typo: missing space

delarosatrevin committed:
Gctf can use now multi GPUs

delarosatrevin committed:
Add angpix_ref, fix when the references have not the same pixel size

delarosatrevin committed:
Merge branch 'release-1.1.facilities-devel' into relion-more-fixes

delarosatrevin committed:
Merge branch 'release-1.1.facilities-fixes' into relion-more-fixes

delarosatrevin committed:
Merged new fixes

Commit:
fixing a CTF parameter assignation (Q0)

CL2D (in fourier) was failing because the input particles came with different Q0 values,
even if the Amplitude Contrast was set at the import. This bug is fix in this commit  in the ctf estimation program.

azazellochg committed:
fix #1530

revert b383e7672aa6bc3991f267bcd084f99b970739f1

pconesa committed:
Merge branch 'release-1.1.facilities-devel' into release-1.1.facilities-fixes

pconesa committed:
Copy info to setOfParticle from setOfMicrographs in extract particles protocols. Relax delta in relion extract particle test for ctf values.

pconesa committed:
Merge pull request #1521 from I2PC/relion-more-fixes

Relion more fixes, rlnRandomSubset propagated in LocalRec

pconesa committed:
Merge branch 'release-1.1.facilities-devel' into release-1.1.facilities-fixes

pconesa committed:
Merge pull request #1531 from I2PC/release-1.1.facilities-fixes

Release 1.1.facilities fixes

Commit:
Updating the version, nickname and releasedate variables to make the bundle

Commit:
Merge branch 'release-1.1.facilities-devel' into release-1.1.facilities-fixes

Commit:
Merge pull request #1534 from I2PC/release-1.1.facilities-fixes

Version, nickname and releasedate updating to make the bundle

azazellochg committed:
hide patch overlap if not using mc2

delarosatrevin committed:
Fixed bug on extract-particles in Relion and the input_micrograph.star

delarosatrevin committed:
Merge branch 'release-1.1.facilities-devel' into release-1.1.facilities-fixes

delarosatrevin committed:
Merge branch 'release-1.1.facilities-fixes' into sdevel

delarosatrevin committed:
Removed debug print

delarosatrevin committed:
Allow to pass references for classification 2D/3D. Also input 3D ref is automatically scaled in accordance with particles

delarosatrevin committed:
Added simple protocol in Relion to center averages (center of mass)

delarosatrevin committed:
Working on Relion export-particles protocol, not finished

delarosatrevin committed:
Finished the implementation of Relion export-particles protocol

delarosatrevin committed:
Allow to pass the calibrated pixel size in Relion postprocess protocol

delarosatrevin committed:
Removed release day

delarosatrevin committed:
Removed duplicated function

delarosatrevin committed:
Moved base picking protocols to a separate file

delarosatrevin committed:
Fixed imports after picking protocols files re-organization

delarosatrevin committed:
Relion autopick working with batch mode

delarosatrevin committed:
Removed debug print

delarosatrevin committed:
Gautomatch can now be run in many GPUs and pick in batches

delarosatrevin committed:
Code formatting

delarosatrevin committed:
Micrographs defocus subset moved to a function

delarosatrevin committed:
Fixed wizard for Gautomatch and allow defocus subset

delarosatrevin committed:
Fixed issue with Gautomatch starting class at 0, but Relion expect at 1

delarosatrevin committed:
Imported missing re

delarosatrevin committed:
Changed label for GPUs

azazellochg commented:
Hi @delarosatrevin , I've started to look at gautomatch.

- [x]  the wizard picks only the first mic in the subset? When I switch to next mic in the picker it does not pick it..
- [x] gautomatch -gid option does not recognize "0 1", since the program does not support multiple GPUs
- [x]  -gid option in gctf is added to the end of cmdline (after micName > output.log), while it should go before those
- [x] [here](https://github.com/delarosatrevin/scipion/blob/sdevel/pyworkflow/em/protocol/protocol_particles_picking.py#L311) help would be good

delarosatrevin commented:
Thanks @azazellochg for taking a look. 

First thing: It can be defined by the provider picker.conf file if you want the wizard to pick all micrographs or just one by one. The latter is the original behavior and the first one I added upon a user request because it is true that for relion make more sense to pick all at once. One question for you, what do you think will be the best for Gautomatch? 

Second, it is a bug that I will fix. I think it was introduced when changing the picker wizard to pick all at once.

azazellochg commented:
@delarosatrevin I think that one-by-one make sense in general (except relion where there is no difference, since it is very slow) since wizard is supposed to be interactive. Otherwise the wizard is kind of useless since running the protocol does the same thing (except displaying results takes a couple more clicks). Another solution could be leave one-by-one picking if user chooses subset=all for the wizard, and all-at-once for a subset=few

azazellochg commented on pyworkflow/em/packages/gautomatch/convert.py:
```diff
@@ -145,38 +144,57 @@ def writeSetOfCoordinates(workDir, coordSet, isGlobal=False):
 _rlnClassNumber #4
 _rlnAutopickFigureOfMerit #5
-"""
+    """
```
minor fix: no need for tab here, otherwise the first line after header is shifted

azazellochg commented on pyworkflow/em/packages/relion/protocol_export_particles.py:
```diff
@@ -0,0 +1,134 @@
+                               "Write multiple stacks",
+                               "Write a single stack"], default=STACK_MULT,
+                      display=params.EnumParam.DISPLAY_LIST,
+                      label="Iteration to visualize",
```
wrong label and help missing

delarosatrevin replied:
Fixed!

delarosatrevin committed:
Centralized method for streaming params and improved help

delarosatrevin committed:
Fixed label and help

delarosatrevin committed:
Fixed gid argument and added some validations

delarosatrevin committed:
Minor validation about batch size

delarosatrevin committed:
Validate threads and GPUs in Gautomatch

delarosatrevin committed:
Centralized function to handle batches, extraction going for batches

delarosatrevin committed:
Centralized sleeping logic for streaming

delarosatrevin committed:
Cleaning up of relion extract particles, now all cases are handled in a general way

delarosatrevin committed:
Allow xmipp picking to specify sleep time and batch (batch not optimized)

delarosatrevin committed:
Stored full hostname to avoid marking as failed protocols launched in other machines

delarosatrevin committed:
Allow to change the user configuration with SCIPION_LOCAL_CONFIG environment variable

delarosatrevin committed:
Improved how input references are handled, addressing sets of 1 vol

delarosatrevin committed:
Fixed bug when changing of micrograph and launching autopicking in wizards

delarosatrevin committed:
Specified MODE_AUTOMATIC in GenericPicker wizards

delarosatrevin committed:
Show defocus in picking GUI if available

delarosatrevin commented:
@azazellochg Thanks for looking into this PR! I fixed the issue with the wizard and also added the possibility to display 'defocusU' in the picking micrograph list (now only for Relion wizard). I want to maybe add resolution and fit and close this PR...what do you think?

azazellochg commented:
@delarosatrevin this is amazing! I wish I could understand java, I wanted to add this years ago :) I will have a look this week.

delarosatrevin commented:
Heheheh...I understand Java, but the logic in the viewers/pickers has become very messy. @airenzp help me with the Java code for the bug fix (also with the wizard changes before) and show me the way to modify the micrograph table. Do you think sorting the columns could be useful? Some here asked because he setup the parameters and check for different defocus. Not sure how easy/difficult would be to implement it.

azazellochg commented:
Sorting columns in picker is the most important thing, that I miss since we added sorting of objects in set selection.

delarosatrevin commented:
Ok, I will do an extra effort. ;)

delarosatrevin committed:
Fixed bug with empty rlnPhaseShift that make export-ctfs crash

delarosatrevin committed:
Fixed bug when scheduling CTF estimation and the project db changes in between
